web/developer.actyx.com/docs/tutorials/getting-started/machine-runner/state-payloads.mdx

```diff
@@ -0,0 +1,348 @@
+  lastMl: 0,
+  totalMl: 0,
+})
+```
+
+<details>
+  <summary>
+    Open to see the full contents of <code>robot.ts</code>
+  </summary>
+
+```ts title="src/robot.ts"
+import { createMachineRunner } from '@actyx/machine-runner'
+import { Actyx } from '@actyx/sdk'
+import { Events, manifest, protocol } from './protocol'
+
+type SpentWater = {
+  lastMl: number
+  totalMl: number
+}
+
+const machine = protocol.makeMachine('robot')
+
+export const Idle = machine.designState('Idle').withPayload<SpentWater>().finish()
+
+export const WateringPlant = machine.designState('WateringPlant').withPayload<SpentWater>().finish()
+
+Idle.react([Events.NeedsWater], WateringPlant, (state, event) => {
+  console.log(`The plant is requesting ${event.payload.requiredWaterMl} ml of water!`)
+  const newStatePayload = {
+    lastMl: event.payload.requiredWaterMl,
+    totalMl: state.self.totalMl + event.payload.requiredWaterMl,
+  }
+  console.log(`Total water consumption: ${newStatePayload.totalMl}`)
+  return WateringPlant.make(newStatePayload)
+})
+
+WateringPlant.react([Events.HasWater], Idle, (state, _) => Idle.make(state.self))
+
+export async function main() {
+  const sdk = await Actyx.of(manifest)
+  const tags = protocol.tagWithEntityId('robot-1')
+  const machine = createMachineRunner(sdk, tags, Idle, {
+    lastMl: 0,
+    totalMl: 0,
+  })
+
+  for await (const state of machine) {
+    console.log(state)
+  }
+}
+
+main()
+```
+
+</details>
+
+## Running our new robot!
+
+Before we run our robot, let's add some logs to track our progress and request water more than once.
+
+On `robot.ts`:
+
+```diff
+  Idle.react([Events.NeedsWater], WateringPlant, (state, event) => {
+    console.log(
+      `The plant is requesting ${event.payload.requiredWaterMl} ml of water!`
+    )
+    const newStatePayload = {
+      lastMl: event.payload.requiredWaterMl,
+      totalMl: state.self.totalMl + event.payload.requiredWaterMl,
+    }
++   console.log(`Total water consumption: ${newStatePayload.totalMl}`)
+    return WateringPlant.make(newStatePayload)
+  })
+```
+
+And on `sensor.ts`:
+
+```diff
+async function main() {
+  const sdk = await Actyx.of(manifest);
+  const tags = protocol.tagWithEntityId("robot-1");
+
++   for (let i = 0; i < 3; i++) {
+      await sdk.publish(
+        tags.apply(
+          Events.NeedsWater.make({
+            requiredWaterMl: Math.floor(Math.random() * 100),
+          })
+        )
+      );
+      console.log("Published NeedsWater");
+
+      await sdk.publish(tags.apply(Events.HasWater.make({})));
+      console.log("Published HasWater");
++   }
+
+  sdk.dispose();
+}
+```
+
+Now we can run the robot and sensor like we have been doing.
+
+<details>
+<summary>You'll see a similar output on your robot (remember that the amount of water is random).</summary>
+
+```bash
+$ npm run start-robot
+
+> tomato-robot@1.0.0 start-robot
+> tsc && node src/index.js
+
+{
+  is: [Function: is],
+  as: [Function: as],
+  cast: [Function: cast],
```

Comment:
To me dumping the whole JS object doesn’t add much value — how about removing `{ is, as, cast }` in the code above, noting that these are functions?